src/ir/module-utils.h

```diff
@@ -38,51 +38,25 @@ struct BinaryIndexes {
   std::unordered_map<Name, Index> eventIndexes;
 
   BinaryIndexes(Module& wasm) {
-    auto addGlobal = [&](Global* curr) {
-      auto index = globalIndexes.size();
-      globalIndexes[curr->name] = index;
-    };
-    for (auto& curr : wasm.globals) {
-      if (curr->imported()) {
-        addGlobal(curr.get());
-      }
-    }
-    for (auto& curr : wasm.globals) {
-      if (!curr->imported()) {
-        addGlobal(curr.get());
-      }
-    }
-    assert(globalIndexes.size() == wasm.globals.size());
-    auto addFunction = [&](Function* curr) {
-      auto index = functionIndexes.size();
-      functionIndexes[curr->name] = index;
-    };
-    for (auto& curr : wasm.functions) {
-      if (curr->imported()) {
-        addFunction(curr.get());
+    auto addIndexes = [&](auto& source, auto& indexes) {
```

Comment:
wait, you can do `auto&` in a lambda parameter..?

Reply:
Yes, thanks C++14!